common/persistence/nosql/nosql_history_store_test.go

```diff
@@ -527,47 +527,630 @@ func TestDeleteHistoryBranch_unusedBranch(t *testing.T) {
 	assert.NoError(t, err)
 }
 
-func TestDeleteHistoryBranch_usedBranch(t *testing.T) {
+// | In this base-case scenario, a workflow's been forked a few times, and the
+// | parent / ancestor workflow (branch A) is being removed.
+// |
+// |  Trees
+// |  ┌───────────────────┐    ┌───────────────────┐   ┌────────────────────┐
+// |  │  Tree: 123        │    │  Tree: 123        │   │ Tree: 123          │
+// |  │  Branch: C        │    │  Branch: A        │   │ Branch: B          │
+// |  │  Ancestors:       │    │                   │   │ Ancestors:         │
+// |  │    Branch: A      │    │                   │   │  Branch A          │
+// |  │    EndNode: 3     │    │                   │   │  EndNode: 2        │
+// |  └───────────────────┘    └───────────────────┘   └────────────────────┘
+// |  Nodes
+// |                           ┌───────────────────┐
+// |                           │     Tree: 123     │
+// |                           │     Branch: A     │
+// |                           │     Node: 1       │
+// |                           └─────────┬─────────┘
+// |                                     │
+// |                                     ┼───────────────────────┐
+// |                           ┌─────────▼─────────┐   ┌─────────▼─────────┐
+// |                           │     Tree: 123     │   │     Tree: 123     │
+// |                           │     Branch: A     │   │     Branch: B     │
+// |                           │     Node: 2       │   │     Node: 2       │
+// |                           └─────────┬─────────┘   └─────────┬─────────┘
+// |            ┌────────────────────────┤                       │
+// |            │                        │                       │
+// |  ┌─────────▼─────────┐    ┌─────────▼─────────┐   ┌─────────▼─────────┐
+// |  │     Tree: 123     │    │     Tree: 123     │   │     Tree: 123     │
+// |  │     Branch: B     │    │     Branch: A     │   │     Branch: B     │
```

Comment:
There are 2 branch Bs. I think the left one should be branch C?

Reply:
Same applies to the other trees.

Reply:
🤦  ty, sorry, I thought I fixed that one.